5.2.cup_of_join.py

```diff
@@ -0,0 +1,12 @@
+import functools
+
+
+def join(*lists, sep='-'):
+    """Concatenate lists using a separator and return as string."""
+    if not lists:
+        return None
+
+    return functools.reduce(lambda x, y: x + [sep] + y, lists)
+
+
+print(join([1, 2], [8], [9, 5, 6]))
```

Comment:
Expressions outside of a function are a bad habit. This code will be called every time the file is loaded and imported.   Read more here: https://www.geeksforgeeks.org/python-main-function/amp/